spatialdata/_core/elements.py

```diff
@@ -84,23 +107,38 @@ def shape(self) -> Tuple[int, ...]:
 
 
 class Labels(BaseElement):
-    def __init__(self, labels: DataArray, transform: Transform) -> None:
-        self.data: DataArray = labels
+    def __init__(self, labels: DataArray, transform: BaseTransformation) -> None:
+        if isinstance(labels, DataArray):
+            self.data = labels
+        elif isinstance(labels, np.ndarray):
+            self.data = DataArray(labels)
+        elif isinstance(labels, dask.array.core.Array):
+            self.data = DataArray(labels)
+        else:
+            raise TypeError("Labels must be a DataArray, numpy array or dask array")
         self.transforms = transform
         super().__init__()
 
     @staticmethod
     def parse_labels(data: Any, transform: Optional[Any] = None) -> "Labels":
-        data, transform = parse_dataset(data, transform)
+        # data, transform = parse_dataset(data, transform)
+        if transform is None:
+            transform = Identity()
         return Labels(data, transform)
 
     def to_zarr(self, group: zarr.Group, name: str, scaler: Optional[Scaler] = None) -> None:
+        assert isinstance(self.transforms, BaseTransformation)
+        coordinate_transformations = self.transforms.to_dict()
+        # at the moment we don't use the compressor because of the bug described here (it makes some tests the
+        # test_readwrite_roundtrip fail) https://github.com/ome/ome-zarr-py/issues/219
         write_labels(
             labels=self.data.data,
             group=group,
             name=name,
             axes=["y", "x"],  # TODO: infer before.
             scaler=scaler,
+            storage_options={"compressor": None},
+            coordinate_transformations=[[coordinate_transformations]],
```

Comment:
same here se compressor and also described above.

Reply:
but also, maybe this can be handled in separate IO PR

Reply:
done